### PR TITLE
Extend `user-local-time` to user profiles

### LINKED ISSUE #5775: Extend `user-local-time` to user profiles
### Description

Quote https://github.com/refined-github/refined-github/pull/2549#issuecomment-559750761:

> * Also display it on profiles like [`linkify-user-location`](https://github.com/refined-github/refined-github/blob/main/source/features/linkify-user-location.tsx)

### Screenshot

image

### Example URLs

https://github.com/fregante

### PR DESCRIPTION
**Pull request by kidonng**
Resolve #5775

A lot of dup code, get this thing out first

## Test URLs

https://github.com/fregante

### CODE REVIEW

File: source/features/user-local-time.tsx [hunk 1]
```diff
@@ -79,6 +79,35 @@ function parseOffset(date: string): number {
 	return (hours * 60) + (hours < 0 ? -minutes : minutes);
 }
 
+async function handleDatePromise({
```
Comment (yakov116) [inline source/features/user-local-time.tsx:82]: How about naming it something with display? Isn't that what this function does?

Comment (kidonng) [reply source/features/user-local-time.tsx:82]: It's just a random name I come up with, these code are so intertwined I have no idea what I'm doing.

**Comment by fregante** [comment]
The structure looks about right. If we had something like Preact it’d be presumably easier to handle this sort of content. 

I wonder if attachElement would make this somewhat easier, where you have an intermediary function which calls attachElement with the time-getter logic inside the getNewElement callback. 

```py 
onHovercardOpen
  append(user, hovercardLocation)

onProfileLoad
  append(user, profileLocation)

append
  attachElement(
    append: location
    getNewElement
      all creation and update logic

**Comment by kidonng** [comment]
> If we had something like Preact it’d be presumably easier to handle this sort of content.

Or SolidJS, which uses native DOM that maybe interesting for our use case. (Not trying to stir up another UI library war, just mentioning 😂)

**Review by fregante — APPROVED** [review]
TL;DReview

**Comment by fregante** [comment]
The problem is that we can’t use JSX (due to the issues described in my last attempt PR) and Solid.js has some footguns when used without it: https://www.solidjs.com/guides/getting-started#buildless-options

Preact doesn’t appear to have this issue. 

I’m not attached to any library but anything that makes it easy, painfree and is under 5kb would be great.

**Comment by fregante** [comment]
I think this would be the correct solution: https://lit.dev/

Unlike most other libraries, these elements are regular dom elements that can be created and appended even with our JSX setup because they’re triggered by a basic `body.append(document.createElement('rgh-time'))` and it automatically detects when the element is removed. 

Any custom-element-based library should work though.

**Comment by kidonng** [comment]
Do we have a list of (relatively) highly dynamic features that will benefit from using a UI library?

**Comment by fregante** [comment]
It’s probably not a whole lot, that’s why library size is important. But if it’s available we’ll surely come up with more. The options page would also definitely benefit from dynamic components, we have a lot of dynamic code in it

**Pull request merged**